### PR TITLE
Chunk

### PR DESCRIPTION
**Pull request by phoniks**

### CODE REVIEW

File: spec/head_spec.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
-    expect(head([1,2,3,4,5,])).toEqual(1)
-    expect(head([2,3,4,5,])).toEqual(2)
-    expect(head([3,4,5,])).toEqual(3)
+    var array = [1,2,3,4,5,]
+    expect(head(array)).toEqual(1)
+    console.log('array: '+array)
+  })
+
+  it('returns the first element in an array', () => {
+    var array2 = [2,3,4,5,]
```
Comment (Moniarchy) [inline spec/head_spec.js]: `const` should be favored over `var`

File: spec/head_spec.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+    expect(head(array2)).toEqual(2)
+    console.log('array2: '+array2)
```
Comment (Moniarchy) [inline spec/head_spec.js]: Remove `console.log`

File: spec/head_spec.js [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
+  })  
+
+  it('returns the first element in an array', () => {
+    var array3 = [3,4,5,]
+    expect(head(array3)).toEqual(3)
+    console.log('array2: '+array3)
```
Comment (Moniarchy) [inline spec/head_spec.js]: Remove

File: spec/head_spec.js [hunk 4]
```diff
@@ -1,10 +1,34 @@
...
+  })
+
+  it('returns the first element in an array', () => {
+    var array2 = [2,3,4,5,]
+    expect(head(array2)).toEqual(2)
+    console.log('array2: '+array2)
+  })  
+
+  it('returns the first element in an array', () => {
+    var array3 = [3,4,5,]
```
Comment (Moniarchy) [inline spec/head_spec.js]: const

File: spec/head_spec.js [hunk 5]
```diff
@@ -1,10 +1,34 @@
...
+    expect(head(array3)).toEqual(3)
+    console.log('array2: '+array3)
+
+  })
+
+  it('returns empty array when passed null', () => {
+    expect(head(null)).toEqual('this')
+    console.log('null yeilds: '+'this')
```
Comment (Moniarchy) [inline spec/head_spec.js]: remove

File: spec/head_spec.js [hunk 6]
```diff
@@ -1,10 +1,34 @@
...
+
+  })
+
+  it('returns the first element in an array of objects', () => {
     expect(head([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual({first: 'first'})
+    console.log('result: '+head([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}]))
```
Comment (Moniarchy) [inline spec/head_spec.js]: remove

File: spec/tail_spec.js [hunk 7]
```diff
@@ -1,10 +1,38 @@
-var tail = require("../src/tail.js")
+import {tail} from '../src/tail'
 
 describe('tail', ()=> {
   it('returns the first element in an array', () => {
-    expect(tail([1,2,3,4,5])).toEqual([2,3,4,5])
-    expect(tail([2,3,4,5,])).toEqual([3,4,5])
-    expect(tail([3,4,5,])).toEqual([4,5])
-    expect(tail([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual([{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}])
+    var array = [1,2,3,4,5]
```
Comment (Moniarchy) [inline spec/tail_spec.js]: const

File: spec/tail_spec.js [hunk 8]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
+    var array2 = [2,3,4,5]
+    expect(tail(array2))
+    .toEqual([3,4,5])
+  }),
+
+  it('returns the first element in an array', () => {
+    var array3 = [3,4,5,]
```
Comment (Moniarchy) [inline spec/tail_spec.js]: const

File: spec/tail_spec.js [hunk 9]
```diff
@@ -1,10 +1,38 @@
...
 
 describe('tail', ()=> {
   it('returns the first element in an array', () => {
-    expect(tail([1,2,3,4,5])).toEqual([2,3,4,5])
-    expect(tail([2,3,4,5,])).toEqual([3,4,5])
-    expect(tail([3,4,5,])).toEqual([4,5])
-    expect(tail([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual([{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}])
+    var array = [1,2,3,4,5]
+    expect(tail(array))
+    .toEqual([2,3,4,5])
```
Comment (Moniarchy) [inline spec/tail_spec.js]: indent, or move to previous line (the latter is preferred since this is such a short line)

File: spec/tail_spec.js [hunk 10]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
+    var array2 = [2,3,4,5]
+    expect(tail(array2))
+    .toEqual([3,4,5])
```
Comment (Moniarchy) [inline spec/tail_spec.js]: indent, or move to previous line (the latter is preferred since this is such a short line)

File: spec/tail_spec.js [hunk 11]
```diff
@@ -1,10 +1,38 @@
...
-    expect(tail([2,3,4,5,])).toEqual([3,4,5])
-    expect(tail([3,4,5,])).toEqual([4,5])
-    expect(tail([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual([{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}])
+    var array = [1,2,3,4,5]
+    expect(tail(array))
+    .toEqual([2,3,4,5])
+  }),
+
+  it('returns the first element in an array', () => {
+    var array2 = [2,3,4,5]
```
Comment (Moniarchy) [inline spec/tail_spec.js]: const

File: spec/tail_spec.js [hunk 12]
```diff
@@ -1,10 +1,38 @@
...
+    expect(tail(array2))
+    .toEqual([3,4,5])
+  }),
+
+  it('returns the first element in an array', () => {
+    var array3 = [3,4,5,]
+    expect(tail(array3))
+    .toEqual([4,5])
```
Comment (Moniarchy) [inline spec/tail_spec.js]: indent, or move to previous line (the latter is preferred since this is such a short line)

File: spec/tail_spec.js [hunk 13]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
+    var array4 = [
```
Comment (Moniarchy) [inline spec/tail_spec.js]: const

File: spec/tail_spec.js [hunk 14]
```diff
@@ -1,10 +1,38 @@
...
+      {first: 'first'},
+      {second: 'second'},
+      {third: 'third'},
+      {fourth: 'fourth'},
+      {fifth: 'fifth'}
+    ]
+    expect(tail(array4))
+    .toEqual([
```
Comment (Moniarchy) [inline spec/tail_spec.js:25]: indent, or move to previous line (the latter is preferred since this is such a short line)

**Comment by Moniarchy** [comment]
@phoniks some comments left inline

File: spec/chunk_spec.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import { chunk } from '../src/chunk'
+
+describe('chunk', ()=> {
+  it('returns array of elements in an array when chunk size is 1', () => {
+    const chunkThis = [1,2,3,4,5]
+    expect(chunk ( chunkThis, 1 ) ).toEqual( [[1],[2],[3],[4],[5]] )
+   })
```
Comment (jrobcodes) [inline spec/chunk_spec.js:7]: Indentation is off

File: spec/chunk_spec.js [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+    const chunkThis2 = [1,2,3,4,5]
+    expect(chunk ( chunkThis2, 2 ) ).toEqual( [[1,2] , [3,4] , [5]] )
+  })
+
+  it('returns array of elements in an array of specified chunk size, with remainder', () => {
+    const chunkThis3 = [1,2,3,4,5]
+    expect(chunk ( chunkThis3, 3 ) ).toEqual( [[1,2,3], [4,5]] )
+  })
+    
+  it('returns array of elements from an object of specified chunk size if ', () => {
```
Comment (jrobcodes) [inline spec/chunk_spec.js:19]: Was trying to parse this test description - what does the last if mean? (Is there something missing?)

File: spec/chunk_spec.js [hunk 17]
```diff
@@ -0,0 +1,39 @@
+import { chunk } from '../src/chunk'
+
+describe('chunk', ()=> {
+  it('returns array of elements in an array when chunk size is 1', () => {
```
Comment (jrobcodes) [inline spec/chunk_spec.js:4]: The language of these tests is a little confusing - `returns array of elements in an array`.  Maybe `returns a new array composed of arrays of elements when...`

File: spec/head_spec.js [hunk 18]
```diff
@@ -1,10 +1,26 @@
-var head = require("../src/head.js")
+import { head } from '../src/head'
 
 describe('head', ()=> {
   it('returns the first element in an array', () => {
-    expect(head([1,2,3,4,5,])).toEqual(1)
-    expect(head([2,3,4,5,])).toEqual(2)
-    expect(head([3,4,5,])).toEqual(3)
+    const array = [1,2,3,4,5,]
```
Comment (jrobcodes) [inline spec/head_spec.js]: I would omit the separate variable declarations here since they're only used once.

File: spec/head_spec.js [hunk 19]
```diff
@@ -1,10 +1,26 @@
...
+    expect(head(array)).toEqual(1)
+  })
+
+  it('returns the first element in an array', () => {
+    const array2 = [2,3,4,5,]
+    expect(head(array2)).toEqual(2)
+  })  
+
+  it('returns the first element in an array', () => {
+    const array3 = [3,4,5,]
```
Comment (jrobcodes) [inline spec/head_spec.js]: Remove trailing comma in array.

File: spec/head_spec.js [hunk 20]
```diff
@@ -1,10 +1,26 @@
...
   it('returns the first element in an array', () => {
-    expect(head([1,2,3,4,5,])).toEqual(1)
-    expect(head([2,3,4,5,])).toEqual(2)
-    expect(head([3,4,5,])).toEqual(3)
+    const array = [1,2,3,4,5,]
+    expect(head(array)).toEqual(1)
+  })
+
+  it('returns the first element in an array', () => {
+    const array2 = [2,3,4,5,]
```
Comment (jrobcodes) [inline spec/head_spec.js]: Remove trailing comma in array.

File: spec/head_spec.js [hunk 21]
```diff
@@ -1,10 +1,26 @@
...
+    expect(head(array2)).toEqual(2)
+  })  
+
+  it('returns the first element in an array', () => {
+    const array3 = [3,4,5,]
+    expect(head(array3)).toEqual(3)
+  })
+
+  it('returns empty array when passed null', () => {
+    expect(head(null)).toEqual('this')
```
Comment (jrobcodes) [inline spec/head_spec.js]: This is passing?

File: spec/head_spec.js [hunk 22]
```diff
@@ -1,10 +1,26 @@
...
+  })
+
+  it('returns the first element in an array of objects', () => {
```
Comment (jrobcodes) [inline spec/head_spec.js:24]: I'm not sure I understand the purpose of this test? How does it differ from the first test?

File: spec/tail_spec.js [hunk 23]
```diff
@@ -1,10 +1,38 @@
...
 
 describe('tail', ()=> {
   it('returns the first element in an array', () => {
-    expect(tail([1,2,3,4,5])).toEqual([2,3,4,5])
-    expect(tail([2,3,4,5,])).toEqual([3,4,5])
-    expect(tail([3,4,5,])).toEqual([4,5])
-    expect(tail([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual([{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}])
+    const array = [1,2,3,4,5]
+    expect(tail(array))
+    .toEqual([2,3,4,5])
```
Comment (jrobcodes) [inline spec/tail_spec.js]: Short statements should remain on one line.

File: spec/tail_spec.js [hunk 24]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
+    const array2 = [2,3,4,5]
+    expect(tail(array2))
+    .toEqual([3,4,5])
```
Comment (jrobcodes) [inline spec/tail_spec.js]: Short statements should remain on one line.

File: spec/tail_spec.js [hunk 25]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
+    const array3 = [3,4,5,]
+    expect(tail(array3))
+    .toEqual([4,5])
```
Comment (jrobcodes) [inline spec/tail_spec.js]: Short statements should remain on one line.

File: spec/tail_spec.js [hunk 26]
```diff
@@ -1,10 +1,38 @@
...
+  }),
+
+  it('returns the first element in an array', () => {
```
Comment (jrobcodes) [inline spec/tail_spec.js:16]: The text of this test is incorrect.

File: spec/tail_spec.js [hunk 27]
```diff
@@ -1,10 +1,38 @@
-var tail = require("../src/tail.js")
+import {tail} from '../src/tail'
 
 describe('tail', ()=> {
   it('returns the first element in an array', () => {
-    expect(tail([1,2,3,4,5])).toEqual([2,3,4,5])
-    expect(tail([2,3,4,5,])).toEqual([3,4,5])
-    expect(tail([3,4,5,])).toEqual([4,5])
-    expect(tail([{first: 'first'},{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'},])).toEqual([{second: 'second'},{third: 'third'},{fourth: 'fourth'},{fifth: 'fifth'}])
+    const array = [1,2,3,4,5]
```
Comment (jrobcodes) [inline spec/tail_spec.js]: We can omit these variable declarations since they are only used once; may as well declare them inline.

File: spec/head_spec.js [hunk 28]
```diff
@@ -1,10 +1,26 @@
-var head = require("../src/head.js")
+import { head } from '../src/head'
 
 describe('head', ()=> {
```
Comment (jrobcodes) [inline spec/head_spec.js:3]: We're missing a test for the case that the array is empty.
